### DISCUSSION TITLE
Sonoff TH16 + MS01 (moisture soil sensor)

### DISCUSSION THREAD

**Discussion by mario-website** [RavenSystem/esp-homekit-devices#1633] [MEPLHAA Scripts]
Hi, anyone can tell me if this will work at all with [MS01](https://aliexpress.com/item/1005003319491526.html) sensor as I believe it might not be supported yet in HAA (I not checked if even is working before flashing to HAA). TH16 working fine with Si7021 (temperature and humidity sensor).
```
{
  "c": { "l": 13 },
  "a": [{
    "t": 1,
    "0": { "r": [{ "g": 12, "v": 0  }] },
    "1": { "r": [{ "g": 12, "v": 1 }] },
    "b": [{ "g": 0 }],
    "s": 5
  }, {
    "t": 24,
    "g": 14,
    "n": 4,
    "j": 2
  }]
}
```
After looked to [WIKI POST](https://github.com/RavenSystem/esp-homekit-devices/wiki/Temperature-and-Humidity-Sensors), I changed "t": 24 (Temperature & Humidity Sensors) to "t": 23 (Humidity Sensor only) and "n": 4 to all values from 1 to 10. But moisture soil is not the same as humidity sensor so this is why it might not working, any help?

**Comment by mario-website** [comment]
...or anyone know which moisture soil might work (I want to have in sprinkler system, on each line, one moisture soil to do for example "if moisture soil is over some % that line will not active” or any other condition)

**Reply by zerodago** [reply]
Hi !! I need the same thing, did you get it to work?
